addons/pharma/functions/fnc_getBloodVolumeChange.sqf

```diff
@@ -102,27 +108,44 @@ if (!isNil {_unit getVariable [QACEGVAR(medical,ivBags),[]]}) then {
     };
 };
 
-// Movement and recovery of interstital fluid 
-private _shiftValue = 0;
-switch (true) do {
-    case (((_ECB + _ECP) > (_ISP * 0.6)) && ((_ECB + _ECP) > 4500)): {
-        // Negative shifts only happen above 4500ml of blood volume, to prevent issues with falling back into arrest/unconsciousness
-        _shiftValue = (2 min ((_ECP + _ECB) - (_ISP * 0.6)));
-        _ECP = _ECP - _shiftValue;
-        _ISP = _ISP + _shiftValue;
-    };
-    case ((_ECB + _ECP) < (_ISP * 0.6)): {
-        _shiftValue = (2 min ((_ISP * 0.6) - (_ECP + _ECB)));
-        _ECP = _ECP + _shiftValue;
-        _ISP = _ISP - _shiftValue;
+// Movement and recovery of interstital fluid and SRBC collection
+private _SRBCChange = 0;
+
+if (_enableFluidShift) then {
+    private _shiftValue = 0;
+    private _defaultShift = false;
+
+    _SRBCChange = if ((_SRBC > 0) && (_ECB < DEFAULT_ECB)) then { 0.5 } else { 0 };
+    _ECB = _ECB + (_SRBCChange * _deltaT);
+    _SRBC = _SRBC - (_SRBCChange * _deltaT);
+
+    switch (true) do {
+        case (((_ECB + _ECP) > (_ISP * 0.6)) && ((_ECB + _ECP) > 4500)): {
+            // Negative shifts only happen above 4500ml of blood volume, to prevent patients from falling back into arrest/unconsciousness
+            _shiftValue = (1 min ((_ECP + _ECB) - (_ISP * 0.6))) * _deltaT;
+
+            _ECP = _ECP - _shiftValue;
+            _ISP = _ISP + _shiftValue;
+        };
+        case ((_ECB + _ECP) < (_ISP * 0.6)): {
+            _shiftValue = (1 min ((_ISP * 0.6) - (_ECP + _ECB))) *_deltaT;
+
+            if (_shiftValue < 0.1) exitWith { _defaultShift = true; };
+
+            _ECP = _ECP + _shiftValue;
+            _ISP = _ISP - _shiftValue;
+        };
+        default {
+            _defaultShift = true;
+        };
     };
-    default {
-        // If no shift is required, fluids begin returning to baseline in both ISP and SRBC volumes
-        _ISP = _ISP + ((DEFAULT_ISP - _ISP) min 2);
-        _SRBC = _SRBC + ((DEFAULT_SRBC - _SRBC) min 1);
+
+    if (_defaultShift) then {
+        _ISP = _ISP + ((((DEFAULT_ISP - _ISP) max -2) min 2) *_deltaT);
+        _SRBC = _SRBC + ((((DEFAULT_SRBC - _SRBC) max -1) min 1) * _deltaT);  
     };
 };
 
-_unit setVariable [QEGVAR(circulation,bodyFluid), [_ECP, _ECB, (_SRBC - _SRBCChange), _ISP, (_ECP + _ECB)], _syncValues];
+_unit setVariable [QEGVAR(circulation,bodyFluid), [_ECB, _ECP, _SRBC, _ISP, (_ECP + _ECB)], _syncValues];
```

Comment:
why changing values placement? was it oversight first?

Reply:
Oversight first